libgnucash/app-utils/gnc-gsettings.c

```diff
@@ -850,3 +868,51 @@ void gnc_gsettings_version_upgrade (void)
     if (cur_maj_min > old_maj_min)
         gnc_gsettings_set_int (GNC_PREFS_GROUP_GENERAL, GNC_PREF_VERSION, cur_maj_min);
 }
+
+
+void gnc_gsettings_block_all (const gchar *text)
+{
+    GSList *node;
+    GSList *remove_copy = NULL;
+
+    PINFO("registered_signals block list before is %d", g_slist_length (registered_signals));
+    for (node = registered_signals; node; node = node->next)
+    {
+        RegisteredSignalInfo *ri = node->data;
+
+        if (g_signal_handler_is_connected (ri->instance, ri->signal_handler))
+            g_signal_handler_block (ri->instance, ri->signal_handler); // block signal_handler
+        else
+            remove_copy = g_slist_prepend (remove_copy, ri); // add for removal
+    }
+
+    if (g_slist_length (remove_copy) > 0)
+    {
+        PINFO("remove_copy length is %d", g_slist_length (remove_copy));
+        for (node = remove_copy; node; node = node->next)
+        {
+            RegisteredSignalInfo *ri = node->data;
+            GSList *found_item = g_slist_find (registered_signals, ri);
+
+            if (found_item)
+                registered_signals = g_slist_remove (registered_signals, ri);
```

Comment:
Why are you removing callbacks here ?

My guess is because you didn't adjust the gnc_gsettings_remove_by_xyz functions to update the registered_signals list opposite to how you update it during gnc_gsettings_register_callback.